internal/runtime/client/client.go

```diff
@@ -276,6 +278,44 @@ func aggregateSuccessfulResponses(aggregatedResponse runtimehooksv1.ResponseObje
 	aggregatedResponse.SetMessage(strings.Join(messages, ", "))
 }
 
+// CallExtensionOption is the interface for configuration that modifies CallExtensionOptions for a CallExtension call.
+type CallExtensionOption interface {
+	// ApplyToOptions applies this configuration to the given CallExtensionOptions.
+	ApplyToOptions(*CallExtensionOptions)
+}
+
+// CallExtensionCacheEntry is a cache entry for the cache that can be used with the CallExtension call via
+// the WithCaching option.
+type CallExtensionCacheEntry struct {
+	CacheKey string
+	Response runtimehooksv1.ResponseObject
+}
+
+// Key returns the cache key of a CallExtensionCacheEntry.
+func (c CallExtensionCacheEntry) Key() string {
+	return c.CacheKey
+}
+
+// WithCaching enables caching for the CallExtension call.
+type WithCaching struct {
+	Cache        cache.Cache[CallExtensionCacheEntry]
+	CacheKeyFunc func(*runtimeregistry.ExtensionRegistration, runtimehooksv1.RequestObject) string
+}
+
+// ApplyToOptions applies WithCaching to the given CallExtensionOptions.
+func (w WithCaching) ApplyToOptions(in *CallExtensionOptions) {
+	in.WithCaching = true
+	in.Cache = w.Cache
+	in.CacheKeyFunc = w.CacheKeyFunc
+}
+
+// CallExtensionOptions contains the options for the CallExtension call.
+type CallExtensionOptions struct {
+	WithCaching  bool
```

Comment:
Having the bool is only for usability right? `o.Cache != nil` would have had the same result?

Reply:
Yeah, just to be more explicit